sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -818,10 +812,35 @@ pi_result piextPlatformCreateWithNativeHandle(pi_native_handle NativeHandle,
   assert(NativeHandle);
   assert(Platform);
 
-  // Create PI platform from the given Level Zero driver handle or retrieve it
-  // from the cache.
   auto ZeDriver = pi_cast<ze_driver_handle_t>(NativeHandle);
-  return getOrCreatePlatform(ZeDriver, Platform);
+
+  pi_uint32 NumPlatforms = 0;
+  pi_result Res = piPlatformsGet(0, nullptr, &NumPlatforms);
+  if (Res != PI_SUCCESS) {
+    return Res;
+  }
+
+  if (NumPlatforms) {
+    std::vector<pi_platform> PlatformCache(NumPlatforms);
```

Comment:
Small nit here ... this isn't really the cache of platforms any more, so I wouldn't use the name "cache" here.  Maybe name it "Platforms" instead?

Reply:
Done.